### PR TITLE
S2t2 fill db

### PR DESCRIPTION
**Pull request by AlisonYoon**

### CODE REVIEW

File: Utilities/DB.php [hunk 1]
```diff
@@ -3,5 +3,124 @@
...
+            $curl = curl_init();
+            curl_setopt_array($curl, array(
+                CURLOPT_URL => $catImgApiUrl,
+                CURLOPT_CUSTOMREQUEST => "GET",
+                CURLOPT_RETURNTRANSFER => true,
+                CURLOPT_HTTPHEADER => array(
+                    'x-api-key: ec254e44-3996-458b-8522-4933954d8fcd'
+                ),
+            ));
+
```
Comment (mallowmew) [inline Utilities/DB.php]: extra newline

File: Utilities/DB.php [hunk 2]
```diff
@@ -3,5 +3,124 @@
...
+            $imgApiResponse = curl_exec($curl);
+            $error = curl_error($curl);
+            curl_close($curl);
+
+            $responseArray = json_decode($imgApiResponse, true);
+            $breedImgs = [];
+
```
Comment (mallowmew) [inline Utilities/DB.php]: extra newline

File: Utilities/DB.php [hunk 3]
```diff
@@ -3,5 +3,124 @@
...
+        $sqlString = implode(',', $sqlArray);
+
+        $sql = $dbconnect->prepare('INSERT INTO `img` (img_src, breed_id) VALUES ' . $sqlString . ';');
+        $sql->execute();
+    }
+}
+
+$test = new DB();
+//$result = $test->fillCatImg();
```
Comment (MotNosniktaw) [inline Utilities/DB.php]: fillCatImg does not return anything. why assign it to a variable?

File: Utilities/DB.php [hunk 4]
```diff
@@ -3,5 +3,124 @@
...
+        curl_close($curl);
+
+        $responseArray = json_decode($response, true);
+        $breedArray = [];
+        $breedNameArray = [];
+        for ($i = 0; $i < count($responseArray); $i++) {
+            array_push($breedArray, $responseArray[$i]["id"]);
+            array_push($breedNameArray, $responseArray[$i]["name"]);
+        }
+        $associativeBreedArray = array_combine($breedArray, $breedNameArray);
```
Comment (mallowmew) [inline Utilities/DB.php:51]: This array combining thing can be cleaned up - try something like `$associativeBreedArray[$responseArray[$i]["id"]] = $responseArray[$i]["name"];` inside the for loop.

File: Utilities/DB.php [hunk 5]
```diff
@@ -3,5 +3,124 @@
...
+        return $associativeBreedArray;
+    }
+
+    public function fillCatBreedToDB()
```
Comment (MotNosniktaw) [inline Utilities/DB.php]: this isnt called anywhere

Comment (AlisonYoon) [reply Utilities/DB.php]: tested and erased

**Review by mallowmew — CHANGES_REQUESTED** [review]

File: Utilities/DB.php [hunk 6]
```diff
@@ -3,5 +3,141 @@
...
+     * Use the breed id to use it in the API url to send the request
+     * Make the response into an array
+     * Return an associative array of breed name as the $key and all the image source url belongs to that breed as $value(which is in array format)
+     * @return array
+     */
+    public function getCatImg()
+    {
+        $catBreeds = $this->getCatBreed();
+        $imgSrcArray = [];
+        foreach ($catBreeds as $id=>$name) {
```
Comment (FelixLetheren) [inline Utilities/DB.php]: Spacing on fat arrow

**Comment by mallowmew** [comment]
`getCatBreed()` is being called a lot, doing extra API calls - consider storing this value and passing it into the functions that need it rather than doing it at the top of every function.

This could be done as part of the cleanup task.

File: Utilities/DB.php (hunk 4 above)
Comment (AlisonYoon) [reply Utilities/DB.php:51]: clean up should be done in s2t3 branch I believe?

File: Utilities/DB.php [hunk 7]
```diff
@@ -3,5 +3,141 @@
...
+                $breedID = $breedIndex + 1;
+                $sqlArray[] = "('$url', $breedID)";
+            }
+        }
+
+        $sqlString = implode(',', $sqlArray);
+
+        $sql = $dbconnect->prepare('INSERT INTO `img` (img_src, breed_id) VALUES ' . $sqlString . ';');
+        $sql->execute();
+    }
```
Comment (MotNosniktaw) [inline Utilities/DB.php:128]: The actual method calls/data flow should be be included
The calls to other methods within methods should be tidied

File: Utilities/DB.php (hunk 1 above)
Comment (AlisonYoon) [reply Utilities/DB.php]: ok

File: Utilities/DB.php (hunk 7 above)
Comment (AlisonYoon) [reply Utilities/DB.php:128]: yeah I agree. This should be done in s2t3 branch I believe

File: Utilities/DB.php (hunk 6 above)
Comment (AlisonYoon) [reply Utilities/DB.php]: ok

File: Utilities/DB.php (hunk 4 above)
Comment (mallowmew) [reply Utilities/DB.php:51]: Yes, if you like

File: Utilities/DB.php [hunk 8]
```diff
@@ -130,4 +126,10 @@ public function fillCatImg()
...
     }
-}
\ No newline at end of file
+}
+# Make a new object and call the methods
+$DBobject = new DB();
+$breeds = $DBobject->getCatBreed();
+$DBobject->fillCatBreedToDB($breeds);
+$catImgArcArray = $DBobject->getCatImg($breeds);
+$DBobject->fillCatImg($breeds, $catImgArcArray);
```
Comment (mallowmew) [inline Utilities/DB.php]: typo `$catImgArcArray` -> `$catImgSrcArray`

**Review by mallowmew — APPROVED** [review]

File: Utilities/DB.php (hunk 8 above)
Comment (AlisonYoon) [reply Utilities/DB.php]: Tom did it! fixed !

**Pull request merged**